### PR TITLE
get FD count for managed disk based on region

### LINKED ISSUE #1509: Enable Setting Fault & Update Domains for ACS
FEATURE REQUEST

Forked ACS-Engine (I believe from Current will update when confirmed)

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
k8s 1.7.2

**What happened**:
North Europe had a storage issue, this inevitably took out the Fault Domain with 2 out of the 3 masters.  The single remaining master did not pick up management as it could not form a quorum

**What you expected to happen**:
There should be an option to set the number of Fault Domains, or provide warnings when a chosen FD count would present operational risks. E.g. with 2 FDs I assume you need 5 or 7 masters (if you can't have a quorum with 2 I am not a etcd expert so don't know!).  This also needs to be in place for agents (as in min number of agents to provide capacity).

**How to reproduce it** (as minimally and precisely as possible):
Ideally by taking out a FD in an Azure Region, probably best not though please? :)

**Anything else we need to know**:
Need to think about Update Domains too.

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Assigns fault domain count for managed disk based on region support. See:

https://github.com/MicrosoftDocs/azure-docs/blob/master/includes/managed-disks-common-fault-domain-region-list.md

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1509

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
get FD count for managed disk based on region
```

### CODE REVIEW

File: pkg/acsengine/template_generator.go [hunk 1]
```diff
@@ -1225,5 +1225,15 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"shellQuote": func(s string) string {
 			return `'` + strings.Replace(s, `'`, `'\''`, -1) + `'`
 		},
+		"GetPlatformFaultDomainCountForManagedDisks": func() int {
+			switch cs.Location {
+			case "eastus", "eastus2", "westus", "centralus", "northcentralus", "southcentralus", "northeurope", "westeurope":
```
Comment (mboersma) [inline pkg/acsengine/template_generator.go]: According to [the doc](https://github.com/MicrosoftDocs/azure-docs/blob/master/includes/managed-disks-common-fault-domain-region-list.md), "canadacentral" also has 3 max fault domains, so it should be in this list.

**Comment by jackfrancis** [comment]
See https://github.com/Azure/acs-engine/pull/3617

**Comment by abelgana** [comment]
Was this released? This enhancement is nowhere to be found.

**Comment by jackfrancis** [comment]
@abelgana No, this PR was closed (not merged).

@mboersma FYI, who may have an update on how far away we are here

**Comment by abelgana** [comment]
Is it safe to do it? We are planning on forking the code and patching it. Thanks

**Comment by mboersma** [comment]
This code works for creating a cluster, but doesn’t properly handle upgrade cases. That’s where it stalled, but I intend to get back to it this week.

Sorry it’s not completed @abelgana. Please feel free to build on this code and submit a PR if it’s urgent or you’ve already started. Otherwise, I’ll pick this up again at the end of this week.

**Comment by fortinbiz** [comment]
w

**Comment by fortinbiz** [comment]
In which context is it not currently suitable for an upgrade? Is there an issue If I upgrade a cluster while retaining the same fault domain count?

**Pull request closed without merging**